sample-apps/discount-functions-sample-app/web/frontend/vite.config.js

```diff
@@ -1,32 +1,55 @@
-import { defineConfig, loadEnv } from 'vite';
+import { defineConfig } from "vite";
+import { dirname } from "path";
+import { fileURLToPath } from "url";
+import react from '@vitejs/plugin-react';
 import path from 'path';
+if (
+  process.env.npm_lifecycle_event === "build" &&
+  !process.env.CI &&
+  !process.env.SHOPIFY_API_KEY
+) {
+  console.warn(
+    "\nBuilding the frontend app without an API key. The frontend build will not run without an API key. Set the SHOPIFY_API_KEY environment variable when running the build command.\n"
+  );
+}
 
-// prettier-ignore
-const INDEX_ROUTE = "^/(\\?.*)?$";
-const API_ROUTE = '^/api/';
-const ENV_KEYS = [
-  'SHOPIFY_API_KEY',
-  'SHOPIFY_ORDER_DISCOUNT_ID',
-  'SHOPIFY_SHIPPING_DISCOUNT_ID',
-  'SHOPIFY_PRODUCT_DISCOUNT_ID',
-];
+const proxyOptions = {
+  target: `http://127.0.0.1:${process.env.BACKEND_PORT}`,
+  changeOrigin: false,
+  secure: true,
+  ws: false,
+};
 
-const root = new URL('.', import.meta.url).pathname;
+const host = process.env.HOST
+  ? process.env.HOST.replace(/https:\/\//, "")
```

Comment:
1) if HOST uses https protocol, strip the protocol;
2) if HOST uses http protocol, leave the protocol;
3) if HOST is null or undefined, use 'localhost'

I'm curious about why we leave the protocol in (2). Is it just default because only `https` protocol causes trouble?

In general, I prefer to use [Web APIs ](https://developer.mozilla.org/en-US/docs/Web/API/URL)(or [Node versions](https://nodejs.org/api/url.html) of them) for this kind of URL validation / manipulation.

Reply:
This is valid, but we will need to investigate, I would create another issue for this, where we can address how we define the HOST. Since this pattern is existing in other configs, I would ship the fix for Firefox and mobile and work on this afterwards.

Reply:
This should have had a [suggestion] prefix, I don't think it's a blocker.